src/harness/reference_models/interference/aggregate_interference.py

```diff
@@ -0,0 +1,366 @@
+  ant_pattern = esc_point['azimuthRadiationPattern']
+
+  # Initialize array of size length of ant_pattern list
+  ant_pattern_gain = np.arange(len(ant_pattern))
+  for i, pattern in enumerate(ant_pattern):
+    ant_pattern_gain[i] = pattern['gain']
+
+  # Get ESC antenna information
+  esc_antenna_info = interf.EscInformation(antenna_height=esc_point['height'],
+                       antenna_azimuth=esc_point['antennaAzimuth'],
+                       antenna_gain=esc_point['antennaGain'],
+                       antenna_pattern_gain=ant_pattern_gain)
+
+  # Get ESC passband 3550-3680 MHz protection channels
+  protection_channels = interf.getProtectedChannels(interf.ESC_LOW_FREQ_HZ, 
+                          interf.ESC_HIGH_FREQ_HZ)
+
+  aggregate_interference.__init__()
+
+  aggregateInterferenceForPoint(protection_point, 
+    protection_channels, interf.ESC_LOW_FREQ_HZ, 
+    interf.ESC_HIGH_FREQ_HZ, grant_list, None, 
+    esc_antenna_info, interf.ProtectedEntityType.ESC)
+  
+  return aggregate_interference.GetAggregateInterferenceInfo() 
+
+
+def calculateAggregateInterferenceForGwpz(gwpz_record, cbsd_list):
+  """Calculates aggregate interference for GWPZ.
+  
+  Calculates aggregate interference for each protection constraint in the GWPZ
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the GWPZ protection entity.
+
+  Args:
+    gwpz_record: GWPZ protection entity
+    cbsd_list: list of CBSD objects containing registrations and grants
+  Returns:
+    Aggregate interference to GWPZ in the nested dictionary format.
+    {latitude : {longitude: [aggr_interf1(mW), aggr_interf2(mW)]}}
+  """
+
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+  gwpz_region = gwpz_record['zone']['features'][0]['properties']['clutter']
+
+  # Get Fine Grid Points for a GWPZ protection area
+  protection_points = utils.GridPolygon(gwpz_record, 2)
+  gwpz_freq_range = gwpz_record['deploymentParam']['operationParam']['operationFrequencyRange']
+  gwpz_low_freq = gwpz_freq_range['lowFrequency']
+  gwpz_high_freq = gwpz_freq_range['highFrequency']
+
+  # Get channels over which area incumbent needs partial/full protection
+  protection_channels = interf.getProtectedChannels(gwpz_low_freq, gwpz_high_freq)
+
+  aggregate_interference.__init__()
+  # Calculate aggregate interference from each protection constraint with a 
+  # pool of parallel processes.
+  pool = Pool(processes=min(multiprocessing.cpu_count(), len(protection_points)))
+  interference = partial(aggregateInterferenceForPoint, channels=protection_channels, 
+                     low_freq=gwpz_low_freq, high_freq=gwpz_high_freq, 
+                     grant_objects=grant_list, fss_info=None, 
+                     esc_antenna_info=None, 
+                     protection_ent_type=interf.ProtectedEntityType.GWPZ_AREA,
+                     region_type=gwpz_region)
+
+  pool.map(interference, protection_points)
+
+  pool.close()
```

Comment:
Remove these 2 lines, this is useless since you are using pool.map() and actually harmful if reusing the pool

Reply:
Updated   as per the suggestion